File: SDKs/amazonaws.com/eventbridge/2015-10-07/go/pkg/models/shared/putrulerequest.go
package shared



type PutRuleRequest struct {
    Description *string `json:"Description,omitempty"`
    EventBusName *string `json:"EventBusName,omitempty"`
    EventPattern *string `json:"EventPattern,omitempty"`
    Name string `json:"Name"`
    RoleArn *string `json:"RoleArn,omitempty"`
    ScheduleExpression *string `json:"ScheduleExpression,omitempty"`
    State *RuleStateEnum `json:"State,omitempty"`
    Tags []Tag `json:"Tags,omitempty"`
    
}

File: SDKs/amazonaws.com/events/2015-10-07/python/src/sdk/models/shared/putrulerequest.py
from dataclasses import dataclass, field
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class PutRuleRequest:
    name: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    description: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Description') }})
    event_bus_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EventBusName') }})
    event_pattern: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EventPattern') }})
    role_arn: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('RoleArn') }})
    schedule_expression: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ScheduleExpression') }})
    state: Optional[RuleStateEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('State') }})
    tags: Optional[List[Tag]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    
